pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -109,14 +109,28 @@ func (r *ApicastReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	disableApicastPortReconcile := "false"
+	apiManagerAnnotations := r.apiManager.GetAnnotations()
+	if apiManagerAnnotations == nil {
+		apiManagerAnnotations = make(map[string]string)
+	}
+	if val, ok := apiManagerAnnotations["apps.3scale.net/disable-apicast-service-reconciler"]; ok {
+		disableApicastPortReconcile = val
+	}
+
+	apicastServiceMutator := reconcilers.CreateOnlyMutator
+	if disableApicastPortReconcile == "false" {
+		apicastServiceMutator = reconcilers.ServicePortMutator
+	}
```

Comment:
```suggestion
	apicastServiceMutator := r.getApiCastServiceMutator()
```

```
func (r ApicastReconciler) getApiCastServiceMutator() reconcilers.MutateFn {
	disableApicastPortReconcile := "false"
	apiManagerAnnotations := r.apiManager.GetAnnotations()
	if apiManagerAnnotations == nil {
		apiManagerAnnotations = make(map[string]string)
	}
	if val, ok := apiManagerAnnotations["apps.3scale.net/disable-apicast-service-reconciler"]; ok {
		disableApicastPortReconcile = val
	}

	apicastServiceMutator := reconcilers.CreateOnlyMutator
	if disableApicastPortReconcile == "false" {
		apicastServiceMutator = reconcilers.ServicePortMutator
	}

	return apicastServiceMutator
}
```

Minor suggestion, but would be nice to put this to it's own function and adding a unit test for testing the returned mutator :thinking:

Reply:
Thanks Kevin, adding this now